external-table/src/gpdbwritableformatter.c

```diff
@@ -481,17 +495,20 @@ gpdbwritableformatter_export(PG_FUNCTION_ARGS)
 	/* Get the number of valid columns, excludes dropped columns */
 	nvalidcolumns = 0;
 	for (i = 0; i < ncolumns; i++)
-		if (!tupdesc->attrs[i]->attisdropped)
-			nvalidcolumns++;
+	{
+            Form_pg_attribute attr = getAttributeFromTupleDesc(tupdesc,i);
+
+            if (!attr->attisdropped)
+                nvalidcolumns++;
+	}
 
 	/*
 	 * Initialize the context structure
 	 */
 	if (myData == NULL)
 	{
-		if (FORMATTER_GET_EXTENCODING(fcinfo) != PG_UTF8)
-			ereport(ERROR, (errcode(ERRCODE_EXTERNAL_ROUTINE_EXCEPTION),
-							errmsg("gpdbwritable formatter can only export UTF8 formatted data. Define the external table with ENCODING UTF8")));
+
+        reportErrorIfNonUTF8Encoding(fcinfo);
```

Comment:
Would it be cleaner if this function was turned into one that returned a boolean?

```suggestion
if (!isUtf8Encoding(fcinfo))
{
    ereport(ERROR, (errcode(ERRCODE_EXTERNAL_ROUTINE_EXCEPTION),
        errmsg(FORMATTER_ENCODING_ERR_MSG, "import")));
}
```